solidity/contracts/Mailbox.sol

```diff
@@ -95,6 +100,14 @@ contract Mailbox is
         _setDefaultIsm(_module);
     }
 
+    /**
+     * @notice Sets the hook if required for the specific ISM
+     * @param _hook The new hook contract
+     */
+    function setHook(address _hook) external onlyOwner {
```

Comment:
Hmm I'd suggest against modifications to the Mailbox for now. Instead, can folks just call the hook themselves?